planner/core/exhaust_physical_plans.go

```diff
@@ -1470,11 +1474,26 @@ func (p *LogicalJoin) exhaustPhysicalPlans(prop *property.PhysicalProperty) ([]P
 		}
 	})
 
+	if prop.IsFlashOnlyProp() && ((p.preferJoinType&preferBCJoin) == 0 && p.preferJoinType > 0) {
+		return nil, false
+	}
+	joins := make([]PhysicalPlan, 0, 5)
```

Comment:
why the max size is 5?

Reply:
I didn't change the original code, the counts of alternatives is larger than 5, but I don't think it's a big deal.

Reply:
index join, index hash join, index merge join, merge join, and hash join, that is the 5?  add broadcast join, so that to 6?

Reply:
yeah, both hash join and broadcast join have two alternatives, so I set it 8